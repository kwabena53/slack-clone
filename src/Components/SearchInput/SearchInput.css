
.searchOrg{
    background: var(--light-white);
    color: white;
    padding: 5px;
    border: none;
    min-width: 0;
    max-width: 732px;
    height: 24px;
    border-radius: 6px;
    flex: 5;
    margin: 0 2em;
}
.searchOrg:hover{
    background: rgb(255, 255, 255, 0.25);
  }


.searchText{
    padding: 0 10px;
    font-size: 13px;
}
