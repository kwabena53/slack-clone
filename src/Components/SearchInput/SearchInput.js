
import "./SearchInput.css"
import { Search } from "../Icons/SlackCloneIcons"

const SearchInput = () => {
    return (
        <button className="searchOrg">
            <Search width="1em" height="1em" color="white" />
            <span className="searchText"> Search Arizona State University</span>
        </button>
    )
}

export default SearchInput