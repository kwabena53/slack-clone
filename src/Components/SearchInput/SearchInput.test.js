import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SearchInput from "./SearchInput";

test("renders search input ", () => {
  render(<SearchInput />);
  const textElement = screen.getByText(/Search Arizona State University/i);
  expect(textElement).toBeInTheDocument();
});

test("matches Search input snapshot", () => {
  const component = renderer.create(<SearchInput/>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

