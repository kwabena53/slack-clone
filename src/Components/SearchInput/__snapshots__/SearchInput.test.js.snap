// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Search input snapshot 1`] = `
<button
  className="searchOrg"
>
  <svg
    className="icon"
    enableBackground="0 0 24 24;"
    height="1em"
    id="Layer_1"
    version="1.1"
    viewBox="0 0 24 24"
    width="1em"
    x="0px"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    y="0px"
  >
    <path
      className="st0"
      d="M11.5,21.8c-5.7,0-10.2-4.6-10.2-10.2S5.8,1.2,11.5,1.2s10.2,4.6,10.2,10.2S17.1,21.8,11.5,21.8z M11.5,2.8 c-4.8,0-8.8,3.9-8.8,8.8s3.9,8.8,8.8,8.8s8.8-3.9,8.8-8.8S16.3,2.8,11.5,2.8z"
    />
    <path
      className="st0"
      d="M22,22.7c-0.2,0-0.4-0.1-0.5-0.2l-2-2c-0.3-0.3-0.3-0.8,0-1.1c0.3-0.3,0.8-0.3,1.1,0l2,2c0.3,0.3,0.3,0.8,0,1.1 C22.4,22.7,22.2,22.7,22,22.7z"
    />
  </svg>
  <span
    className="searchText"
  >
     Search Arizona State University
  </span>
</button>
`;
